test/syscalls/linux/pty.cc

```diff
@@ -1148,6 +1148,23 @@ TEST_F(PtyTest, SwitchNoncanonToCanonNoNewlineBig) {
   ExpectFinished(replica_);
 }
 
+// If the canonical input buffer is empty, and we switch to non-canonical
+// mode, the input buffer should not be readable.
+TEST_F(PtyTest, SwitchCanonToNoncanonEmptyInput) {
+  constexpr char kInput[] = "";
+  ASSERT_THAT(WriteFd(master_.get(), kInput, sizeof(kInput)),
+              SyscallSucceedsWithValue(sizeof(kInput)));
+
+  DisableCanonical();
+
+  // Nothing available yet.
+  char buf[2] = {};
+  ASSERT_THAT(PollAndReadFd(replica_.get(), buf, 2, kTimeout),
```

Comment:
`kTimeout` is 20 seconds. In the success case, this test will hang for 20 seconds? Its too long. For such successful timeouts, we should use a smaller value like 2 seconds.

Reply:
Makes sense. Is there a constant (like `kSmallTimeout`) I should use, or should I just hardcode it as 2?

Reply:
I think hard coding here is fine.